File: class_7/class_7/app/src/main/java/com/alejouribesanchez/class_7/Main6Activity.java
package com.alejouribesanchez.class_7;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.CheckBox;
import android.widget.LinearLayout;

public class Main6Activity extends AppCompatActivity {

    public static String[] INDUSTRIAS = {
            "Arte",
            "Computación",
            "Ingeniería civil",
            "Bioquímica",
            "Música",
            "Astronomía",
            "Zoología"
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main6);

        LinearLayout seccionIndustria = (LinearLayout) findViewById(R.id.activity_main6);

        for (String industria : INDUSTRIAS) {
            CheckBox opcion = new CheckBox(this);
            opcion.setText(industria);
            opcion.setLayoutParams(
                    new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT));
            seccionIndustria.addView(opcion);
        }

        View layout = (View)findViewById(R.id.activity_main6);
        layout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(Main6Activity.this, Main7Activity.class));
            }
        });
    }
}

File: class_7/class_7/app/src/main/java/com/alejouribesanchez/class_7/Main6Activity.kt
package com.alejouribesanchez.class_7

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.CheckBox
import android.widget.LinearLayout

class Main6Activity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main6)

        val seccionIndustria = findViewById(R.id.activity_main6) as LinearLayout

        for (industria in INDUSTRIAS) {
            val opcion = CheckBox(this)
            opcion.text = industria
            opcion.layoutParams = LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT)
            seccionIndustria.addView(opcion)
        }

        findViewById(R.id.activity_main6).setOnClickListener { startActivity(Intent(this@Main6Activity, Main7Activity::class.java)) }
    }

    companion object {

        var INDUSTRIAS = arrayOf("Arte", "Computación", "Ingeniería civil", "Bioquímica", "Música", "Astronomía", "Zoología")
    }
}
